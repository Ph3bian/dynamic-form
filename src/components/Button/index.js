import React from "react";
import styles from "./module.scss";

const Button = ({ type, children, text, onClick, variant, ...rest }) => (
  <button
    type={type || "button"}
    onClick={onClick}
    className={[styles.Button, styles[variant]].join(" ")}
    {...rest}
  >
    {text || children}
  </button>
);
export default Button;
