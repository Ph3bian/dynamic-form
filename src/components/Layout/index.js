import React from 'react'
import styles from './layout.module.scss'
import {Italy } from 'assets/svg'
import ErrorBoundary from "../ErrorBoundary"
export const Layout = ({ children }) => (
 
  <div className={styles.Layout}>
    <div className={styles.LayoutBg}>
      <h1>Welcome</h1>
      <Italy />
    </div>
    <div className={styles.LayoutContent}> {children}</div>
  </div>
 
)
export default Layout
