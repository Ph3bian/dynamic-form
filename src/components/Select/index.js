import React from "react";
import styles from "./select.module.scss";
const Select = ({ options, defaultValue, label, ...rest }) => {
  return (
    <div className={styles.Select} tabIndex={0}>
      <label htmlFor={`listbox-${label}`}>{label}</label>
      <select
        id={`listbox-${label}`}
        value={defaultValue}
        {...rest}
        data-testid={'TextSelect'}
       
      >
        <option>Select {label}</option>
        {options &&
          options.length > 0 &&
          options.map((option) => (
            <option key={option} value={option}>
              {option}
            </option>
          ))}
      </select>
    </div>
  );
};
export default Select;
