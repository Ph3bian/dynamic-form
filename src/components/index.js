import Button from "./Button";
import Input from "./Input";
import Select from "./Select";
import Layout from "./Layout";
export { Button, Input, Select,Layout };
