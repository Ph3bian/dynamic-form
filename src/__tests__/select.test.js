import React from "react";

import { render, fireEvent } from "@testing-library/react";
import { Select } from "../../components";

describe("The <Select /> component", () => {
  const defaultProps = {
    onChange: jest.fn(),
    name: "firstName",
    options: ["Phebian", "Anwuli"],
    label: "Name",
    defaultValue: "",
  };

  test("Should render select correctly", () => {
    const { asFragment } = render(<Select {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("Should call the onChange handler when it is provided", () => {
    const { getByTestId } = render(<Select {...defaultProps} />);
    const selectField = getByTestId("TextSelect");
    const inputValue = "Anwuli";

    fireEvent.change(selectField, { target: { value: inputValue } });
    expect(defaultProps.onChange).toHaveBeenCalled();
  });
});
