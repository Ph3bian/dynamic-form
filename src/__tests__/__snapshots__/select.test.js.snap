// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <Select /> component Should render select correctly 1`] = `
<DocumentFragment>
  <div
    class="Select"
    tabindex="0"
  >
    <label
      for="listbox-Name"
    >
      Name
    </label>
    <select
      data-testid="TextSelect"
      id="listbox-Name"
      name="firstName"
    >
      <option>
        Select Name
      </option>
      <option
        value="Phebian"
      >
        Phebian
      </option>
      <option
        value="Anwuli"
      >
        Anwuli
      </option>
    </select>
  </div>
</DocumentFragment>
`;
